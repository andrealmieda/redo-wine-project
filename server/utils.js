const path = require('path');
const requireFromString = require('require-from-string');

// Constants
const JS_FILE = /^.+\.js$/;
const CSS_FILE = /^.+\.css$/;

const parseManifestFile = buffer => JSON.parse(buffer);

const parseServerSource = buffer => requireFromString(buffer.toString()).default;

const createAssetRealPath = assetName => path.join(__dirname, '../dist/', assetName);

const createAssetServedPath = assetName => `/assets/${assetName}`;

const normalizeAssetsForTemplate = (manifest) => {
    const assets = {
        js: [],
        css: [],
    };

    Object.values(manifest).reduce((assetsAccumulator, asset) => {
        const { js, css } = assetsAccumulator;

        if (JS_FILE.test(asset)) {
            js.push(createAssetServedPath(asset));
        } else if (CSS_FILE.test(asset)) {
            css.push(createAssetServedPath(asset));
        }

        return assetsAccumulator;
    }, assets);

    return assets;
};

module.exports = {
    createAssetRealPath,
    createAssetServedPath,
    normalizeAssetsForTemplate,
    parseManifestFile,
    parseServerSource,
};
