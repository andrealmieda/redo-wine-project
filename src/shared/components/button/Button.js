import React from 'react';
import PropTypes from 'prop-types';


import styled from 'styled-components';
import { rgba } from 'polished';

import { colors, matchBreakpoint } from '../../styles';

/* eslint-disable react/button-has-type */
const Button = styled(({ primary, secondary, ...props }) => <button {...props} />)`
    margin: 5px;
    padding: ${({ primary, secondary }) => (primary || secondary ? '10px' : '0px')};
    
    background: ${({ primary }) => (primary ? colors.NightSlatz : colors.White)};
    color: ${({ primary }) => (primary ? colors.White : colors.NightSlatz)};
    
    border: ${({ secondary }) => (secondary ? `1px solid ${colors.NightSlatz};` : 'none')};
    border-radius: 5px;
`;
/* eslint-enable react/button-has-type */

Button.propTypes = {
    primary: PropTypes.bool,
    secondary: PropTypes.bool,
};

export default Button;
