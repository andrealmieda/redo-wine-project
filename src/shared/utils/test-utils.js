import React from 'react';

import { mount as enzymeMount } from 'enzyme';
import { merge } from 'lodash-es';

export const configureMount = ({ defaultProps = {}, defaultMount = enzymeMount }) => {
    const mountComponent = (Component, { props = {}, mount = defaultMount, mergeProps = true } = {}) => {
        const componentProps = mergeProps ? merge({}, defaultProps, props) : props;

        return mount(<Component {...componentProps} />);
    };

    return mountComponent;
};

export const createDataTestSelector = dataTest => `[data-test]=${dataTest}`;

export const getElementsByDataTest = (tree, dataTest) => {
    const elementSelector = createDataTestSelector(dataTest);

    return tree.find(elementSelector);
};
