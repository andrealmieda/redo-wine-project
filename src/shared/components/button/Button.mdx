---
name: Button
route: /components/button
order: 1
parent: Components
menu: Components
---

import Button from './Button.js'
import { Playground } from 'docz';

# Button

This component is a styled version of the `button` component from react. It doesn't change any of the original props.
This component supports 3 different flavors. 

- Default
- Primary
- Secondary

To use the flavors just pass the props `primary` or `secondary` accordingly to `<Button />`. 
The default flavor is active unless one of the other flavors props is passed  

## Visual look
Here is the look of the flavors `primary`, `secondary` and default respectively.
<Playground>
    <Button primary> Hello I'm the Primary </Button>
    <Button secondary> Secondary here! </Button>
    <Button> I'm the default one </Button>
</Playground>
