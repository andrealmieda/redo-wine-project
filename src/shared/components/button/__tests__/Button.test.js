import Button from '../index';
import { configureMount } from '../../../utils/test-utils';

const defaultProps = {
    children: 'Hello ðŸ‘‹ I am the button text',
}
const mountComponent = configureMount({ defaultProps })

describe('<Button />', () => {
    it('should render correctly', () => {
        const tree =  mountComponent(Button);
        
        expect(tree).toMatchSnapshot();
    });

    it('shouldn\'t pass style related props to the button', () => {
        const props = {
            primary: true,
            secondary: true,
        }
        const tree =  mountComponent(Button, {props});
        const button = tree.find('button');

        expect(button.props).not.toContain(props)
        
        expect(tree).toMatchSnapshot();
    })
});
