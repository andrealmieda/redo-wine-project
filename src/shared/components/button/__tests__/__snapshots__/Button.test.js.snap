// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
.c0 {
  margin: 5px;
  padding: 0px;
  background: #ffffff;
  color: #a52b4c;
  border: none;
  border-radius: 5px;
}

<Styled(Component)>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
    margin: 5px;
    padding: ",
            [Function],
            ";
    
    background: ",
            [Function],
            ";
    color: ",
            [Function],
            ";
    
    border: ",
            [Function],
            ";
    border-radius: 5px;
",
          ],
        },
        "displayName": "Styled(Component)",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "primary": [Function],
          "secondary": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
  >
    <Component
      className="c0"
    >
      <button
        className="c0"
      >
        Hello ðŸ‘‹ I am the button text
      </button>
    </Component>
  </StyledComponent>
</Styled(Component)>
`;

exports[`<Button /> shouldn't pass style related props to the button 1`] = `
.c0 {
  margin: 5px;
  padding: 10px;
  background: #a52b4c;
  color: #ffffff;
  border: 1px solid #a52b4c;
  border-radius: 5px;
}

<Styled(Component)
  primary={true}
  secondary={true}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
    margin: 5px;
    padding: ",
            [Function],
            ";
    
    background: ",
            [Function],
            ";
    color: ",
            [Function],
            ";
    
    border: ",
            [Function],
            ";
    border-radius: 5px;
",
          ],
        },
        "displayName": "Styled(Component)",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "primary": [Function],
          "secondary": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
    primary={true}
    secondary={true}
  >
    <Component
      className="c0"
      primary={true}
      secondary={true}
    >
      <button
        className="c0"
      >
        Hello ðŸ‘‹ I am the button text
      </button>
    </Component>
  </StyledComponent>
</Styled(Component)>
`;
